core/templates/pages/community-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -39,26 +39,42 @@ type ContributionOpportunityParams = {
   language_code?: string;
 };
 
-export interface IFetchOpportunitySuccessCallbackParams {
-  opportunities?: SkillOpportunity[],
+export interface IFetchedOpportunitiesSuccessCallbackParams {
+  opportunities?: IOpportunityDict[],
   nextCursor?: string,
   more?: boolean
 }
 
+export interface IOpportunityDict {
```

Comment:
@h-ARTS, is this the object it refers to: `/core/templates/domain/opportunity/ExplorationOpportunitySummaryObjectFactory.ts`

Reply:
It supposes to be for both (`SkillOpportunityObjectFactory` and `ExplorationOpportunitySummaryObjectFactory`) due to `IFetchedOpportunitiesSuccessCallbackParams` as it uses for the single value in `.then()` callback. 

If I use a union type like the one below the linter will show this error:
```
export interface IFetchedOpportunitiesSuccessCallbackParams {
  opportunities: SkillOpportunity[] | ExplorationOpportunitySummary[],
  nextCursor: string,
  more: boolean
}
```
<img width="849" alt="Screenshot 2020-05-08 at 11 48 35" src="https://user-images.githubusercontent.com/11369764/81394585-710fd500-9122-11ea-8b0d-9663aea88abc.png">

Reply:
@DubeySandeep PTAL

Reply:
I would suggest creating separate interface for each so that we can use it in the domain class as well, will that be fine (Also, move this to the domain class and remove `any` for the dict object there.)

Reply:
@DubeySandeep Sorry for the late reply, PTAL